manic/repository_git.py

```diff
@@ -103,7 +103,25 @@ def _clone_repo(self, base_dir_path, repo_dir_name, verbosity):
         """
         cwd = os.getcwd()
         os.chdir(base_dir_path)
-        self._git_clone(self._url, repo_dir_name, verbosity)
+
+        # get url
+        url = self._url
+
+        # add port
+        if self._port is not EMPTY_STR:
+            ind1 = url.find('//')+2
+            ind2 = url[ind1:].find('/')
+            url = url[:ind1+ind2] + ':' + self._port + url[ind1+ind2:]
```

Comment:
Is this always the right logic for where in the url the port should be added, or would it potentially need to be generalized? I really don't know the answer to this myself, but if you think it might need to be generalized in the future, then please add a comment about that.